src/telegram/process_update_task.rs

```diff
@@ -25,15 +26,90 @@ impl ProcessUpdateTask {
         Self { update }
     }
 
-    pub async fn process(&self) {
-        // TODO: process users messages here
-        // Create a ApiClient
+    pub async fn process(&self) -> Result<(), BotError> {
+        log::info!("Received a message {:?}", self.update);
+
         let api = AsyncApi::new(&RUST_TELEGRAM_BOT_TOKEN);
-        let opwm_token = &OPEN_WEATHER_MAP_API_TOKEN;
+        // for now with _ to pass warnings
+        let _opwm_token = &OPEN_WEATHER_MAP_API_TOKEN;
```

Comment:
why are you assigning it to a variable if you're not using it?

Reply:
I will delete

Reply:
Done